evap/results/tools.py

```diff
@@ -202,6 +202,30 @@ def average_non_grade_rating_questions_distribution(results):
     )
 
 
+def calculate_average_course_distribution(course):
+    if course.evaluations.exclude(state="published").exists():
+        return None
```

Comment:
the if checks whether any unpublished evaluation exists. if a course has unpublished evaluations, no result will be calculated.

Reply:
i see that, but why do you need the special case? shouldn't the code below return some empty result in that case all by itself?
also, isn't it inconsistent that if all evaluations are unpublished, nothing is calculated, but as soon as there is one published evaluation, the unpublished ones are included in the calculation as well?

Reply:
i think we're talking at cross purposes, because i don't get what you're asking at all.
if one evaluation is published, nothing is done, only if all evaluations are published, results are calculated.